src/ophyd_async/epics/pmac/_util.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+from ophyd_async.core._utils import gather_dict
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.pmac._pmac_io import PmacIO
+
+
+@dataclass
+class _PmacMotorInfo:
+    cs_port: str
+    cs_number: int
+    motor_cs_index: dict[Motor, int]
+    motor_acceleration_rate: dict[Motor, float]
+
+    @classmethod
+    async def from_motors(
+        cls, pmac: PmacIO, motors: Sequence[Motor]
+    ) -> "_PmacMotorInfo":
```

Comment:
If we
```python
from __future__ import annotations
```
Then this can become

```suggestion
    ) -> _PmacMotorInfo:
```